neurons/validators/utils/utils.py

```diff
@@ -32,4 +36,30 @@ def count_hotkeys_per_ip(filtered_axons):
         ip = axon.ip
         hotkey_count_per_ip[ip] = hotkey_count_per_ip.get(ip, 0) + 1
 
-    return hotkey_count_per_ip
\ No newline at end of file
+    return hotkey_count_per_ip
+
+def generate_challenge_to_check(node, start_block_height, last_block_height, k=20):
+    blocks_to_check = random.sample(range(start_block_height, last_block_height + 1), k=k)
+    txn_ids_to_check = []
+    graph_creator = GraphCreator()
+    challenge_inputs = []
+    
+    for block_height in blocks_to_check:
+        block_data = node.get_block_by_height(block_height)
+        num_transactions = len(block_data["tx"])
+
+        out_total_amount = 0
+        while out_total_amount == 0:
+            selected_txn = block_data["tx"][random.randint(0, num_transactions - 1)]
+            txn_id = selected_txn.get('txid')
+        
+            txn_data = node.get_txn_data_by_id(txn_id)
+            tx = graph_creator.create_in_memory_txn(txn_data)
+
+            in_amount_by_address, out_amount_by_address, input_addresses, output_addresses, in_total_amount, out_total_amount = process_in_memory_txn_for_indexing(tx, node)
```

Comment:
the name of the function is a bit misleading as validator will not index anything

Reply:
I wonder if we could have on BitcoinNode and Node in general a `node.generate_challenge()` function that would return a `ChallengeInput`. 
We must ensure that the code for validator remains agnostic of the network (btc, eth etc)